pyomo/solvers/plugins/solvers/MIPCL.py

```diff
@@ -0,0 +1,167 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import logging
+
+import pyutilib.services as services
+
+from pyutilib.common import ApplicationError
+from pyutilib.misc import Bunch, Options
+from pyutilib.services.TempfileManager import create_tempfile
+
+from pyomo.util.plugin import alias
+from pyomo.opt import SolverResults
+from pyomo.opt.base import OptSolver, SolverFactory, ProblemFormat, ResultsFormat
+from pyomo.opt.solver import SystemCallSolver
+from pyomo.opt.base.solvers import _extract_version
+from pyomo.solvers.mockmip import MockMIP
+
+from six import iteritems, string_types
+
+logger = logging.getLogger('pyomo.solvers')
+
+_mipcl_version = None
+
+def configure_mipcl():
+    global _mipcl_version
+    if _mipcl_version is not None:
+        return
+    _mipcl_version = _extract_version('') #TODO
+    if services.registered_executable('mps_mipcl') is None:
+        return
+    
+class MIPCL(OptSolver):
+    """The MIPCL LP/MIP solver"""
+    
+    alias('mipcl', doc='The MIPCL LP/MIP solver')
+    
+    def __new__(cls, *args, **kwds):
+        configure_mipcl()
+        try:
+            mode = kwds['solver_io']
+            if mode is None:
+                mode = 'mps'
+            del kwds['solver_io']
+        except KeyError:
+            mode = 'mps'
+        if mode  == 'mps':
+            opt = SolverFactory('_mipcl_shell', **kwds)
+            opt.set_problem_format(ProblemFormat.mps)
+            return opt
+        if mode  == 'mod':
+            opt = SolverFactory('_mipcl_shell', **kwds)
+            opt.set_problem_format(ProblemFormat.cpxlp)
+            return opt
+        if mode == 'os':
+            opt = SolverFactory('_ossolver', **kwds)
+        else:
+            logger.error('Unknown IO type: %s' % mode)
+            return
+        opt.set_options('solver=mps_mipcl')
+        return opt
+
+class MIPCLSHELL(SystemCallSolver):
+    """Shell interface to the MIPCL LP/MIP solver"""
+    
+    alias('_mipcl_shell', doc='Shell interface to the MIPCL LP/MIP solver')
+    
+    def __init__(self, **kwds):
+        kwds['type'] = 'mipcl'
+        SystemCallSolver.__init__(self, **kwds)
+        
+        self._valid_problem_formats = [ProblemFormat.mps, ProblemFormat.mod]
```

Comment:
I've never seen the 'mod' format used by a solver plugin. Maybe someone can fill me in on how this would work.